File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/dto/fetchoptions/FetchOptions.js
define([ "stjs", "dto/fetchoptions/CacheMode" ], function(stjs, CacheMode) {
	var FetchOptions = function() {
		this.count = null;
		this.from = null;
		this.cacheMode = CacheMode.NO_CACHE;
	};
	stjs.extend(FetchOptions, null, [], function(constructor, prototype) {
		prototype['@type'] = 'dto.fetchoptions.FetchOptions';
		constructor.serialVersionUID = 1;
		prototype.count = function(count) {
			this.count = count;
			return this;
		};
		prototype.getCount = function() {
			return this.count;
		};
		prototype.from = function(from) {
			this.from = from;
			return this;
		};
		prototype.getFrom = function() {
			return this.from;
		};
		prototype.cacheMode = function(cacheMode) {
			this.cacheMode = cacheMode;
			return this;
		};
		prototype.getCacheMode = function() {
			return this.cacheMode;
		}
	}, {});
	return FetchOptions;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/as/api/v3/dto/common/fetchoptions/FetchOptions.java
package ch.ethz.sis.openbis.generic.as.api.v3.dto.common.fetchoptions;

import java.io.Serializable;

import ch.systemsx.cisd.base.annotation.JsonObject;

@JsonObject("dto.common.fetchoptions.FetchOptions")
public abstract class FetchOptions<OBJECT> implements Serializable
{

    private static final long serialVersionUID = 1L;

    private Integer count;

    private Integer from;

    private CacheMode cacheMode = CacheMode.NO_CACHE;

    @SuppressWarnings("hiding")
    public FetchOptions<OBJECT> count(Integer count)
    {
        this.count = count;
        return this;
    }

    public Integer getCount()
    {
        return count;
    }

    @SuppressWarnings("hiding")
    public FetchOptions<OBJECT> from(Integer from)
    {
        this.from = from;
        return this;
    }

    public Integer getFrom()
    {
        return from;
    }

    public FetchOptions<OBJECT> cacheMode(CacheMode mode)
    {
        this.cacheMode = mode;
        return this;
    }

    public CacheMode getCacheMode()
    {
        return cacheMode;
    }

    public abstract SortOptions<OBJECT> sortBy();

    public abstract SortOptions<OBJECT> getSortBy();

}
